plugins/woocommerce/includes/abstracts/abstract-wc-product.php

```diff
@@ -1659,7 +1660,7 @@ protected function is_visible_core() {
 	 * @return bool
 	 */
 	public function is_purchasable() {
-		return apply_filters( 'woocommerce_is_purchasable', $this->exists() && ( 'publish' === $this->get_status() || current_user_can( 'edit_post', $this->get_id() ) ) && '' !== $this->get_price(), $this );
+		return apply_filters( 'woocommerce_is_purchasable', $this->exists() && ( ProductStatus::PUBLISH === $this->get_status() || current_user_can( 'edit_post', $this->get_id() ) ) && '' !== $this->get_price(), $this );
```

Comment:
I realize it isn't new as of this PR but, since we're changing this line, we should probably add the hook docblock to satisfy the linter.